### ISSUE TITLE
Question: How to load fast big flat files ?

### ISSUE THREAD

**Issue by healiseu** [yandex/ClickHouse#3784]
The best method I found is to use the table function `file(path, format, structure)`. It takes as an input parameter the relative path to the file from the setting `user_files_path`. One can change this setting in `/etc/clickhouse-server/config.xml` 

**Question:** Is it possible to change `user_files_path` in a clickhouse-client session with an `sql` command ?

I suppose an alternative method instead of copying/placing the flat-file under `user_files_path` is to pipe the flat-file to command line client (`clickhouse-client`) but that requires access to the file system and the command has to be invoked from my python application.

Is there another method to load fast big flat-files ?

**Comment by alexey-milovidov** [comment]
> Question: Is it possible to change user_files_path in a clickhouse-client session with an sql command ?

No, because the setting is security limit: to allow to read only restricted subset of files from server's filesystem. But you can specify `user_files_path` as `/` if you really don't care.

> I suppose an alternative method instead of copying/placing the flat-file under user_files_path is to pipe the flat-file to command line client (clickhouse-client) but that requires access to the file system and the command has to be invoked from my python application.

If your files are on client side and you need to transfer it over the network, better to use `clickhouse-client`. If your files are already on server's filesystem, better to use `file` table function.

Note: sometimes using clickhouse-client may be faster, because it use "double buffering". It will parse next chunk of data while waiting for server to insert previous chunk of data. But this should not be very significant.

> Is there another method to load fast big flat-files (millions of rows) ?

There are also some advanced solutions, like to prepare a partition for MergeTree table, then move it to the server and attach. It should not be faster unless you can do data preparation on separate cluster in parallel.
